web/src/components/vis/Heatmap.vue

```diff
@@ -47,16 +47,38 @@ function aggregate(arr, rs, cs, transposed) {
 const DENDOGRAM_RATIO = 0.2;
 const LABEL_WIDTH = 150;
 
-const MDI_PLUS_CIRCLE = '&#xF417;';
-const MDI_MINUS_CIRCLE = '&#xF376;';
-const MDI_STAR_CIRCLE = '&#xF4CF;';
+const MDI_PLUS_CIRCLE = '\uF417;';
+const MDI_MINUS_CIRCLE = '\uF376;';
+const MDI_STAR_CIRCLE = '\uF4CF;';
 
 export const heatmapLayouts = [
   { label: 'Auto', value: 'auto' },
   { label: 'Square Cells', value: 'squareCells' },
   { label: 'Square Matrix', value: 'squareMatrix' },
 ];
 
+function changeHovered(wrapper, arr) {
+  if (!arr) {
+    if (wrapper.hovered.size > 0) {
+      wrapper.hovered = new Set();
+      return true;
+    }
+    return false;
+  }
+
+  const old = wrapper.hovered;
+
+  if (old.size !== arr.length) {
+    wrapper.hovered = new Set(arr);
+    return true;
+  }
+  if (old.size === 0 || ((arr.length === 1 && old.has(arr[0])) || arr.every(d => old.has(d)))) {
```

Comment:
Could you add a comment about what this does?

Reply:
just compares the array with the set with optimizations for 0 length and length 1

Reply:
Please add some documentation on this function.